zdm/survey.py

```diff
@@ -509,12 +513,59 @@ def __init__(self, state, survey_name:str,
         # Efficiency: width_method passed through "self" here
         pwidths,pprobs=make_widths(self, state)
         _ = self.get_efficiency_from_wlist(dmvals,
-                                       pwidths,pprobs,model=width_bias) 
+                                       pwidths,pprobs,model=width_bias, edir=edir) 
 
     def init_DMEG(self,DMhalo):
         """ Calculates extragalactic DMs assuming halo DM """
         self.DMhalo=DMhalo
         self.DMEGs=self.DMs-self.DMGs-DMhalo
+        # self.DMEGs_obs=self.DMs-self.DMGs-DMhalo
+        # self.DMEGs = np.copy(self.DMEGs_obs)
+        # self.DMEGs[self.DMEGs < 0] = 10. # Minimum value of 10. pc/cm^3
+    
+    def init_zs(self):
+        """Gets zlist and nozlist and determines which z values to use"""
```

Comment:
See comments. Keep this functions, but we need to alter how it is used

Reply:
Btw, have you checked this works when there are no FRBs in the file? I've been tripped up by this before...